onnx/backend/test/case/node/__init__.py

```diff
@@ -21,15 +22,69 @@
 _NodeTestCases = []
 _TargetOpType = ""
 
-from onnx.onnx_pb import NodeProto, AttributeProto, TypeProto, FunctionProto
+from onnx.onnx_pb import NodeProto, AttributeProto, TypeProto, FunctionProto, GraphProto
+
+
+def _rename_edges_helper(internal_node: NodeProto,
+                         rename_helper: Callable[[Text], Text],
+                         attribute_map: Dict[Text, Text],
+                         prefix: Text) -> NodeProto:
+    new_node = NodeProto()
+    new_node.CopyFrom(internal_node)
+    new_node.ClearField("input")
+    new_node.ClearField("output")
+    new_node.ClearField("attribute")
+    for internal_name in internal_node.input:
+        new_node.input.append(rename_helper(internal_name))
+    for internal_name in internal_node.output:
+        new_node.output.append(rename_helper(internal_name))
+    for attr in internal_node.attribute:
+        if attr.HasField("ref_attr_name"):
+            if attr.ref_attr_name in attribute_map:
+                new_attr = AttributeProto()
+                new_attr.CopyFrom(attribute_map[attr.ref_attr_name])  # type: ignore
+                new_attr.name = attr.name
+        else:
+            new_attr = AttributeProto()
+            new_attr.CopyFrom(attr)
+            if attr.type == AttributeProto.GRAPH:
+                new_graph = new_attr.g
+                sg_rename = {}
+                for in_desc in new_graph.input:
+                    sg_rename[in_desc.name] = in_desc.name = prefix + in_desc.name
+                for out_desc in new_graph.output:
+                    sg_rename[out_desc.name] = out_desc.name = prefix + out_desc.name
+                for init_desc in new_graph.initializer:
+                    sg_rename[init_desc.name] = init_desc.name = prefix + init_desc.name
+                for sparse_init_desc in new_graph.sparse_initializer:
+                    sg_rename[sparse_init_desc.values.name] = sparse_init_desc.values.name = prefix + \
+                        sparse_init_desc.values.name
+                for sparse_init_desc in new_graph.sparse_initializer:
+                    sg_rename[sparse_init_desc.indices.name] = sparse_init_desc.indices.name = prefix + \
+                        sparse_init_desc.indices.name
+                for val_info_desc in new_graph.value_info:
+                    sg_rename[val_info_desc.name] = val_info_desc.name = prefix + val_info_desc.name
```

Comment:
The change looks good, thanks! A minor suggestion: I think it would be better to omit `new_graph.output` as well as `new_graph.value_info` above. The reason is to distinguish between "name binding occurrences" which can introduce a new name (for which updating `sg_rename` makes sense) from "references to names defined/bound elsewhere".

Reply:
Done